### PR TITLE
SDL 0236 Update mismatch in TireStatus structure

### LINKED ISSUE #1297: [SDL 0236] Update mismatch in TireStatus structure
Proposal: [Update mismatch in TireStatus structure](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0236-TireStatus-Mismatch.md)

> This proposal is about correcting a mismatch between HMI_API and MOBILE_API for the `TireStatus` structure.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/743

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revision: clarify that this update will be a breaking change.

> The proposal .md file was updated to reflect these revisions on 6/6/19.

### PR DESCRIPTION
**Pull request by yoooriii**
Fixes #1297 

### Risk
This PR makes **major** API change.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behaviour

#### Unit Tests
Updated the correspondent test SDLTireStatusSpec

#### Core Tests
[List of tests performed against Core and behaviors verified]

Core version / branch / commit hash / module tested against: [Core: sdl_0236_tireStatus_mismatch](https://github.com/LuxoftSDL/sdl_core/tree/feature/sdl_0236_tireStatus_mismatch)
HMI name / version / branch / commit hash / module tested against: [HMI: develop](https://github.com/smartdevicelink/sdl_hmi/tree/develop)

### Summary
Added '__nullable' attribute to mark properties optional
Removed the tag 'Required' from comments.

##### Breaking Changes
* no breaking changes

##### Enhancements
* no enhancements

##### Bug Fixes
* no bug fixes

### Tasks Remaining:
- [x] none

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yoooriii** [comment]
Hello @atiwari9, @mrapitis. Could you please review the PR? Thank you.

**Comment by atiwari9** [comment]
@standa1-ford  - Please take note.

**Comment by perryneal11** [comment]
@yoooriii This PR makes changes to the mobile API, can the risk section be updated?

**Comment by perryneal11** [comment]
@yoooriii Could we update the issue fixed to #1297 ?

**Comment by yoooriii** [comment]
> @yoooriii This PR makes changes to the mobile API, can the risk section be updated?

Description updated to: it makes MINOR change

**Comment by perryneal11** [comment]
The first line of the PR references "fixes #1867", shouldn't it say, "fixes #1297 "?

File: SmartDeviceLink/public/SDLTireStatus.h [hunk 1]
```diff
@@ -19,49 +19,49 @@ NS_ASSUME_NONNULL_BEGIN
...
 
  Required
  */
-@property (strong, nonatomic) SDLSingleTireStatus *innerLeftRear;
+@property (strong, nonatomic, nullable) SDLSingleTireStatus *innerLeftRear;
 
 /**
  The status of the innter right rear tire.
 
  Required
```
[the remark this answers is not in the document]
Comment (yoooriii) [reply SmartDeviceLink/public/SDLTireStatus.h]: Done.

**Comment by yoooriii** [comment]
> The first line of the PR references "fixes #1867", shouldn't it say, "fixes #1297 "?

Fixed.

**Comment by perryneal11** [comment]
@yoooriii I was able to run the tests successfully. 

The proposal lists this as a major change. Additionally, the author has this listed as a breaking change. So It should be notated here. The tasks remaining box should be checked as well.

**Comment by yoooriii** [comment]
Let it be a major as the proposal claims.

**Comment by a-prykho** [comment]
@perryneal11 we`ve updated description , could you please re-review this PR and approve?

**Comment by joeljfischer** [comment]
Hi @yoooriii thanks for catching that this is a breaking (major) version change. We will need to discuss internally whether to make the next version of Core / app libs a major version change (which was not approved by the committee) or to pull this proposal from the next version.

**Comment by perryneal11** [comment]
@theresalech This is approved by ford

**Comment by LitvinenkoIra** [comment]
@theresalech This PR is ready for Livio review. Thank you!

**Comment by theresalech** [comment]
@LitvinenkoIra can you please confirm that Ford has approved the latest changes? It looks like their last approval was in January, prior to the most recent commits. Thank you!

**Comment by LitvinenkoIra** [comment]
@perryneal11 Could you please confirm your approval?

**Comment by rbrigh25** [comment]
@theresalech @LitvinenkoIra 
Approved by Ford

**Review by joeljfischer — CHANGES_REQUESTED** [review]

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m [hunk 2]
```diff
@@ -21,57 +21,103 @@
 SDLSingleTireStatus* tire4 = [[SDLSingleTireStatus alloc] init];
 SDLSingleTireStatus* tire5 = [[SDLSingleTireStatus alloc] init];
 SDLSingleTireStatus* tire6 = [[SDLSingleTireStatus alloc] init];
+// make all tires different to prevent misplacement (eg left front != right front etc.)
```
[the remark this answers is not in the document]
Comment (YuriiLokhmatovGitHub) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m:24]: Hello @joeljfischer. I have updated the PR as requested. Would you please review it? Thank you.

**Comment by YuriiLokhmatovGitHub** [comment]
Hello @joeljfischer. I have updated the PR as requested. Would you please review it? Thank you.

**Comment by YuriiLokhmatovGitHub** [comment]
Please note SDLVoiceCommandManagerSpec has nothing to do with this PR and this is the only test that fails.

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m [hunk 3]
```diff
@@ -21,57 +21,103 @@
...
+tire1.pressure = @(1.0);
+tire2.pressure = @(2.0);
+tire3.pressure = @(3.0);
+tire4.pressure = @(4.0);
+tire5.pressure = @(5.0);
+tire6.pressure = @(6.0);
```
[the remark this answers is not in the document]
Comment (joeljfischer) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m:30]: This was not done

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m (hunk 2 above)
[the remark this answers is not in the document]
Comment (joeljfischer) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m:24]: This was not done

**Review by joeljfischer — CHANGES_REQUESTED** [review]
See the comments from the unit tests that were not done.

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m (hunk 3 above)
[the remark this answers is not in the document]
Comment (YuriiLokhmatovGitHub) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m:30]: _Only assign an initial value._ -- If I got it right the 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 are not initial values and must be assigned in a beforeEach block, right?

[the remark this answers is not in the document]
Comment (joeljfischer) [reply SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLTireStatusSpec.m:30]: Yes, that’s correct

**Comment by YuriiLokhmatovGitHub** [comment]
Hello @joeljfischer. I fixed the unit test. Could you please review it? Thank you.

**Review by joeljfischer — APPROVED** [review]

**Pull request merged**